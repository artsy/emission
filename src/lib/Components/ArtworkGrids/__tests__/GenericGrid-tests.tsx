import React from "react"
import "react-native"
import { renderWithLayout } from "../../../tests/renderWithLayout"

import RelayGenericArtworksGrid, { GenericArtworksGrid } from "../GenericGrid"

it("renders properly", () => {
  const artworks = [artwork(), artwork(), artwork()]

  const layout = { width: 768 }

  const grid = renderWithLayout(<RelayGenericArtworksGrid artworks={artworks as any} />, layout)
  expect(grid).toMatchSnapshot()
})

it("handles showing an update when there are new artworks", () => {
  const artworks = [artwork(), artwork()] as any
  const newArtworks = [artwork(), artwork(), artwork()] as any

  const grid = new GenericArtworksGrid({ artworks })
  const shouldUpdate = grid.shouldComponentUpdate({ artworks: newArtworks }, {} as any)

  expect(shouldUpdate).toBeTruthy()
})

const artwork = () => {
  return {
    __id: "artwork-long-title",
    id: "long-title",
    title: "DO WOMEN STILL HAVE TO BE NAKED TO GET INTO THE MET. MUSEUM",
    date: "2012",
    sale_message: null,
    is_in_auction: false,
    image: {
      url: "artsy.net/image-url",
      aspect_ratio: 2.18,
    },
    artists: [
      {
        name: "Guerrilla Girls",
      },
    ],
    partner: {
      name: "Whitechapel Gallery",
    },
    href: "/artwork/guerrilla-girls-do-women-still-have-to-be-naked-to-get-into-the-met-museum",
  }
}
