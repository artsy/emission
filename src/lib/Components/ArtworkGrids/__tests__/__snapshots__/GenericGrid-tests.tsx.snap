// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    accessibilityLabel="Artworks Content View"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityLabel="Section 0"
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "marginRight": 20,
            "width": 242.66666666666666,
          },
        ]
      }
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <AROpaqueImageView
            aspectRatio={2.18}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 8,
              }
            }
          />
        </View>
        <View
          flexDirection="row"
          mb="2px"
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 2,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#000"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Medium"
            fontSize="13px"
            lineHeight="18px"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 13,
                  "lineHeight": 18,
                },
                undefined,
              ]
            }
          >
             
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#666"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Regular"
            fontSize="12px"
            lineHeight="16px"
            style={
              Array [
                Object {
                  "color": "#666",
                  "fontFamily": "Unica77LL-Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Array [
                Object {
                  "color": "#666666",
                  "fontSize": 14,
                  "height": 18,
                  "lineHeight": 18,
                },
                Object {
                  "fontWeight": "bold",
                },
              ],
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Guerrilla Girls
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 14,
                    "height": 18,
                    "lineHeight": 18,
                  },
                  Object {
                    "fontStyle": "italic",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            DO WOMEN STILL HAVE TO BE NAKED TO GET INTO THE MET. MUSEUM
          </Text>
          , 2012
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Whitechapel Gallery
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Section 1"
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "marginRight": 20,
            "width": 242.66666666666666,
          },
        ]
      }
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <AROpaqueImageView
            aspectRatio={2.18}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 8,
              }
            }
          />
        </View>
        <View
          flexDirection="row"
          mb="2px"
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 2,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#000"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Medium"
            fontSize="13px"
            lineHeight="18px"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 13,
                  "lineHeight": 18,
                },
                undefined,
              ]
            }
          >
             
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#666"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Regular"
            fontSize="12px"
            lineHeight="16px"
            style={
              Array [
                Object {
                  "color": "#666",
                  "fontFamily": "Unica77LL-Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Array [
                Object {
                  "color": "#666666",
                  "fontSize": 14,
                  "height": 18,
                  "lineHeight": 18,
                },
                Object {
                  "fontWeight": "bold",
                },
              ],
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Guerrilla Girls
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 14,
                    "height": 18,
                    "lineHeight": 18,
                  },
                  Object {
                    "fontStyle": "italic",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            DO WOMEN STILL HAVE TO BE NAKED TO GET INTO THE MET. MUSEUM
          </Text>
          , 2012
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Whitechapel Gallery
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Section 2"
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "marginRight": 0,
            "width": 242.66666666666666,
          },
        ]
      }
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <AROpaqueImageView
            aspectRatio={2.18}
            imageURL="artsy.net/image-url"
            style={
              Object {
                "marginBottom": 8,
              }
            }
          />
        </View>
        <View
          flexDirection="row"
          mb="2px"
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 2,
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#000"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Medium"
            fontSize="13px"
            lineHeight="18px"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 13,
                  "lineHeight": 18,
                },
                undefined,
              ]
            }
          >
             
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            className="sc-bdVaJa bDWFJH"
            color="#666"
            ellipsizeMode="tail"
            fontFamily="Unica77LL-Regular"
            fontSize="12px"
            lineHeight="16px"
            style={
              Array [
                Object {
                  "color": "#666",
                  "fontFamily": "Unica77LL-Regular",
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                undefined,
              ]
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Array [
                Object {
                  "color": "#666666",
                  "fontSize": 14,
                  "height": 18,
                  "lineHeight": 18,
                },
                Object {
                  "fontWeight": "bold",
                },
              ],
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Guerrilla Girls
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Array [
                  Object {
                    "color": "#666666",
                    "fontSize": 14,
                    "height": 18,
                    "lineHeight": 18,
                  },
                  Object {
                    "fontStyle": "italic",
                  },
                ],
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            DO WOMEN STILL HAVE TO BE NAKED TO GET INTO THE MET. MUSEUM
          </Text>
          , 2012
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "color": "#666666",
                "fontSize": 14,
                "height": 18,
                "lineHeight": 18,
              },
              Object {
                "fontFamily": "AGaramondPro-Regular",
              },
            ]
          }
        >
          Whitechapel Gallery
        </Text>
      </View>
    </View>
  </View>
</View>
`;
